dart/dynamics/Entity.h

```diff
@@ -189,6 +198,19 @@ class Entity : public virtual common::Subject
   bool needsAccelerationUpdate() const;
 
 protected:
+
+  /// Used when constructing a Frame class, because the Frame constructor will
+  /// take care of setting up the parameters you pass into it
+  enum ConstructFrame_t { ConstructFrame };
+
+  explicit Entity(ConstructFrame_t);
```

Comment:
This seems like specialized constructor using enum for specific cases. In this case, it's to avoid calling Entity::changeParentFrame() before Frame::changeParentFrame() is called, right? Is this known technique?